distribution/openrct2.d.ts

```diff
@@ -3150,6 +3151,118 @@ declare global {
         postMessage(message: ParkMessageDesc): void;
     }
 
+    interface Research {
+        /**
+         * The list of rides and scenery sets that have already been researched.
+         */
+        inventedItems: ResearchItem[];
+
+        /**
+         * The order of rides and scenery sets to be researched.
+         */
+        uninventedItems: ResearchItem[];
+
+        /**
+         * The amount of funding currently spent on research.
+         */
+        funding: ResearchFundingLevel;
+
+        /**
+         * Flags representing which research categories are enabled.
+         */
+        priorities: number;
+
+        /**
+         * The current stage for the ride or scenery set being researched.
+         */
+        stage: ResearchFundingStage;
+
+        /**
+         * The progress for the current stage between 0 and 65535.
+         * This will increment more quickly the higher the research funding.
+         */
+        progress: number;
+
+        /**
+         * The expected month the current item being researched will complete.
+         * Value is between 0 and 7, 0 being March and 7 being October.
+         * Value is null if there is not yet an expected month.
+         */
+        readonly expectedMonth: number | null;
+
+        /**
+         * The expected day of the month the current item being researched will complete.
+         * Value is between 0 and 30, add 1 to it for the human readable date.
+         * Value is null if there is not yet an expected month.
+         */
+        readonly expectedDay: number | null;
+
+        /**
+         * Gets whether a particular object has been researched and is available to construct.
+         * @param type The type of object, e.g. ride, scenery group, or small scenery.
+         * @param index The object index.
+         */
+        isObjectResearched(type: ObjectType, index: number): boolean;
+    }
+
+    type ResearchItem = RideResearchItem | SceneryResearchItem;
+
+    interface RideResearchItem {
+        type: "ride";
+
+        /**
+         * The research category this item belongs in.
+         * E.g. gentle rides, thrill rides, shops etc.
+         */
+        category?: ResearchCategory;
```

Comment:
Why is this optional?

Reply:
I am not allowing plugins to override which category a research item belongs in. It is determined by the ride type.

Reply:
Maybe it's a good idea to reflect this behaviour in the comments above it (always present on get, ignored on set), and also make the field `readonly`.

Reply:
Making it `readonly` only works if every field were to be readonly. Should every field be readonly here?

Reply:
If there's no use setting them, then yes.